src/controller/python/chip-device-ctrl.py

```diff
@@ -964,15 +965,24 @@ def __format_zcl_arguments_from_dict(optional_args: dict, command: dict) -> Dict
     return formatted_command_args
     
 
-def zcl_add_network(node_id: int, ssid: str, password: str, endpoint_id: Optional[int] = 1, group_id: Optional[int] = 0, breadcrumb: Optional[int] = 0, timeoutMs: Optional[int] = 1000) -> Dict[str, Any] :
+def zcl_add_network(node_id: int, args: Dict[str, str], endpoint_id: Optional[int] = 1, group_id: Optional[int] = 0, breadcrumb: Optional[int] = 0, timeoutMs: Optional[int] = 1000) -> Dict[str, Any] :
     try:
         __check_supported_os()
-        args = {}
-        args['ssid'] = ssid.encode("utf-8") + b'\x00'
-        args['credentials'] = password.encode("utf-8") + b'\x00'
-        args['breadcrumb'] = breadcrumb
-        args['timeoutMs'] = timeoutMs 
-        err, res = device_manager.devCtrl.ZCLSend("NetworkCommissioning", "AddWiFiNetwork", node_id, endpoint_id, group_id, args, blocking=True)
+        command: str = ""
+        parameters = {}
+        if args.get("ssid") and args.get("password"):
+            parameters['ssid'] = args.get("ssid").encode("utf-8") + b'\x00'
+            parameters['credentials'] = args.get("password").encode("utf-8") + b'\x00'
+            command = "AddWifiNetwork"
+        elif args.get("operationalDataset"):
+            parameters['operationalDataset'] = bytes.fromhex(args.get("operationalDataset"))
+            command = "AddThreadNetwork"
+        else:
+            raise Exception("(ssid and password) or dataset need to be provided")
+        parameters['breadcrumb'] = breadcrumb
+        parameters['timeoutMs'] = timeoutMs
+        err, res = device_manager.devCtrl.ZCLSend("NetworkCommissioning", command, node_id, endpoint_id, group_id,
+                                                  parameters, blocking=True)
```

Comment:
lines too long

Reply:
abbreviations should be avoided unless necessary for long names, use `error, result` instead.